### ISSUE TITLE
为什么最终的模型中要把 prefix_embeds 和 caption_embeds 拼在一起丢进 GPT

### ISSUE THREAD

**Issue by hj24** [yangjianxin1/ClipCap-Chinese#9]
刚学习nlp,可能是一个小白问题, caption 不是标签数据吗? 为什么需要一起丢进模型
# caption_inputs_embeds:[bs, caption_len, prefix_size]
caption_embeds = self.gpt2.transformer.wte(caption_ids)
# prefix_embeds:[bs, prefix_len, prefix_size]
prefix_embeds = self.clip_project(clip_embeds).view(-1, self.prefix_len, self.prefix_size)
# embedding_cat:[bs, prefix_len+caption_len, prefix_size]
embedding_cat = torch.cat((prefix_embeds, caption_embeds), dim=1)
out = self.gpt2(inputs_embeds=embedding_cat, attention_mask=mask)

**Comment by XieYingchun** [comment]
I think that during training, gpt2 needs to have labels.
